docs/tutorial/tutorial-5/android.rst

```diff
@@ -2,9 +2,382 @@
+to run the app on an Android device, either physical or virtual. This section
+covers running the app on a virtual Android device.
+
+On Android, Briefcase's ``run`` command requires that you specify which device to run on.
+If you use ``run`` without specifying a device, Briefcase will tell you how to
+create an appropriate Android virtual device. Run that command on your system.
+
+.. tabs::
+
+  .. group-tab:: macOS
+
+    .. code-block:: bash
+
+      (beeware-venv) $ briefcase run android
+
+      No Android device was specified. Please specify a specific device on which
+      to run the app by passing `-d <device_id>`.
+
+      ...
+
+      If you do not see any devices, you can create and start an emulator by running:
+
+      ...
+
+  .. group-tab:: Linux
+
+    .. code-block:: bash
+
+      (beeware-venv) $ briefcase run android
+
+      No Android device was specified. Please specify a specific device on which
+      to run the app by passing `-d <device_id>`.
+
+      ...
+
+      If you do not see any devices, you can create and start an emulator by running:
+
+      ...
+
+  .. group-tab:: Windows
+
+    .. code-block:: doscon
+
+      C:\...>briefcase run android
+
+      No Android device was specified. Please specify a specific device on which
+      to run the app by passing `-d <device_id>`.
+
+      ...
+
+      If you do not see any devices, you can create and start an emulator by running:
+
+      ...
+
+
+Examine the specific output on your system, and run the commands provided
+until you see an Android virtual device on your screen. It will look something
+like this as it starts up.
+
+.. figure:: ../images/android/tutorial-5-booting.png
+   :alt: Android virtual device booting
+
+   Android virtual device booting
+
+.. figure:: ../images/android/tutorial-5-running.png
+   :alt: Android virtual device fully started
+
+   Android virtual device fully started
+
+Once an Android virtual device is running, you can launch the app with Briefcase
+by running this command:
+
+.. tabs::
+
+  .. group-tab:: macOS
+
+    .. code-block:: bash
+
+      (beeware-venv) $ briefcase run android -d emulator-5554
+
+  .. group-tab:: Linux
+
+    .. code-block:: bash
+
+      (beeware-venv) $ briefcase run android -d emulator-5554
+
+  .. group-tab:: Windows
+
+    .. code-block:: doscon
+
+      C:\...>briefcase run android -d emulator-5554
+
+At this point, you should see the app launch on your virtual device. If you see
+an error message, you may need to follow the advice in the error message about
+finding the correct device name.
+
+Run the app on a physical device
+================================
+
+If you have an Android phone or tablet you want to run your app on, you can
+use Briefcase's ``run`` command if you enable debugging on your device and
+pass its device ID to ``-d <device_id>``.
+
+To find the device name, use the ``run`` command without ``-d`` parameter.
```

Comment:
This isn't really the best way to find the device ID in the circumstances. Given that the user has just gone through the simulator, we can tell them to invoice ADB directly.

Also, need to add instructions to plug the device in first!

Reply:
Fair enough. I'll have to spell out the full adb path, e.g., `$HOME/.briefcase/tools/android_sdk/platform-tools/adb devices -l` , but that's OK.

Reply:
Hmm, the reason I wrote it this way is to avoid typing out the full path to `adb`. In the future, we intend to support using the Android SDK embedded into Android Studio, in which case the `adb` path I'm hard-coding in the docs will become wrong.

Let's discuss before I make changes to this section to be sure we get the best change.